core/src/main/scala/org/ensime/indexer/IndexService.scala

```diff
@@ -84,13 +84,19 @@ class IndexService(path: File) {
 
   private val lucene = new SimpleLucene(path, analyzers)
 
-  def persist(check: FileCheck, symbols: List[FqnSymbol], commit: Boolean): Unit = {
+  def persist(check: FileCheck, symbols: List[FqnSymbol], commit: Boolean, boost: Boolean): Unit = {
     val f = Some(check)
     val fqns: List[Document] = symbols.map {
       case FqnSymbol(_, _, _, fqn, Some(_), _, _, _, _) => MethodIndex(fqn, f).toDocument
       case FqnSymbol(_, _, _, fqn, _, Some(_), _, _, _) => FieldIndex(fqn, f).toDocument
       case FqnSymbol(_, _, _, fqn, _, _, _, _, _) => ClassIndex(fqn, f).toDocument
     }
+    // I don't like casting here, but the alternative is some messy param threading through the
+    // serialization code.
+    // The enormous boost is to ensure we always return user FQNs first
+    if (boost)
+      fqns foreach { _.boostText("fqn", 1000f) }
```

Comment:
I agree. I wasn't sure how extreme to go so started pretty high. I'll drop it down to 10 & see how it looks during a local QA.